llvm/test/LTO/AMDGPU/closed-world-assumption.ll

```diff
@@ -0,0 +1,12 @@
+; RUN: opt -S -mtriple=amdgcn-amd-amdhsa -O3 -debug-only=amdgpu-attributor -o - %s 2>&1 | FileCheck %s --check-prefix=NO-CW
+; RUN: opt -S -mtriple=amdgcn-amd-amdhsa -passes="lto<O3>" -debug-only=amdgpu-attributor -o - %s 2>&1 | FileCheck %s --check-prefix=NO-CW
+; RUN: opt -S -mtriple=amdgcn-amd-amdhsa -passes="lto<O3>" -debug-only=amdgpu-attributor -amdgpu-link-time-closed-world=1 -o - %s 2>&1 | FileCheck %s --check-prefix=CW
+
+; REQUIRES: amdgpu-registered-target
+; REQUIRES: asserts
+
+; NO-CW: Module {{.*}} is not assumed to be a closed world.
+; CW: Module {{.*}} is assumed to be a closed world.
+define hidden noundef i32 @_Z3foov() {
```

Comment:
Does this not have any testable effects? I would expect to see a test that shows the before and after of the expected changes.

Reply:
`NO-CW` and `CW`, isn't it the difference?

Reply:
Yeah but that's just a debug message, I would expect to see IR changes.

Reply:
that is tested in other files. this PR is just to toggle the switch on/off. it doesn't introduce this as a new feature.

Reply:
It's still better to check an actual change, and avoid dependent on asserts builds